### PR TITLE
Reconstruct GPX extensions from OBF-GPX tracks

### PR DESCRIPTION
**Pull request by RZR-UA**
Related PR https://github.com/osmandapp/OsmAnd-tools/pull/884

### CODE REVIEW

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd/issues/18866

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java [hunk 1]
```diff
@@ -1112,6 +1119,14 @@ public boolean publish(Amenity amenity) {
 												String tag = key.replaceFirst(GPX_EXTENSION_TAG_PREFIX, "");
 												String val = amenity.getAdditionalInfo(key);
 												gpxFileExtensions.put(tag, val);
+											} else if (key.startsWith(POINTS_GROUPS_TAG_PREFIX)) {
+												final String DELIMITER = "~~~";
```
Comment (ivanPyrohivskyi) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:1123]: DELIMITER in generation and Android must be same constant (Kotlin lib)

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java [hunk 2]
```diff
@@ -1112,6 +1119,14 @@ public boolean publish(Amenity amenity) {
...
+												String joinedValues = amenity.getAdditionalInfo(key);
+												List<String> values = Arrays.asList(joinedValues.split(DELIMITER));
+												if (key.endsWith("names")) pgNames.addAll(values);
```
Comment (ivanPyrohivskyi) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:1126]: formatting

File: OsmAnd/src/net/osmand/core/android/MapRendererContext.java [hunk 3]
```diff
@@ -205,7 +206,7 @@ private void updateMapPresentationEnvironment(boolean forceUpdateProviders) {
 		}
 		if (!mapStyles.containsKey(rendName)) {
 			Log.d(TAG, "Style '" + rendName + "' not in cache");
-			if (mapStylesCollection.getStyleByName(rendName) == null) {
+			if (mapStylesCollection.getStyleByName(rendName) == null || IGNORE_CORE_PRELOADED_STYLES) {
```
Comment (ivanPyrohivskyi) [inline OsmAnd/src/net/osmand/core/android/MapRendererContext.java:209]: Unnecessary

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java [hunk 4]
```diff
@@ -1097,12 +1108,28 @@ public boolean isCancelled() {
...
+											if (key.startsWith(GPX_EXTENSION_TAG_PREFIX)) {
+												String tag = key.replaceFirst(GPX_EXTENSION_TAG_PREFIX, "");
+												String val = amenity.getAdditionalInfo(key);
+												gpxFileExtensions.put(tag, val);
+											} else if (key.startsWith(POINTS_GROUPS_TAG_PREFIX)) {
+												final String DELIMITER = "~~~";
+												String joinedValues = amenity.getAdditionalInfo(key);
+												List<String> values = Arrays.asList(joinedValues.split(DELIMITER));
+												if (key.endsWith("names")) pgNames.addAll(values);
+												else if (key.endsWith("icons")) pgIcons.addAll(values);
```
Comment (ivanPyrohivskyi) [inline OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:1127]: points_groups_names must be in constant in Kotlin lib

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java (hunk 1 above)
Comment (RZR-UA) [reply OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:1123]: moved to PointsGroups in Kotlin

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java (hunk 2 above)
Comment (RZR-UA) [reply OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:1126]: formatting is OK here (this is internal class instance in method parameters so formatting is looking sligtly weird)

File: OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java (hunk 4 above)
Comment (RZR-UA) [reply OsmAnd/src/net/osmand/plus/wikivoyage/data/TravelObfHelper.java:1127]: done

File: OsmAnd/src/net/osmand/core/android/MapRendererContext.java (hunk 3 above)
Comment (RZR-UA) [reply OsmAnd/src/net/osmand/core/android/MapRendererContext.java:209]: Still required until the corresponding changes would be committed to the core library.

**Review by ivanPyrohivskyi — APPROVED** [review]

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/util/KAlgorithms.kt [hunk 5]
```diff
@@ -190,4 +190,17 @@ object KAlgorithms {
 		}
 	}
 
+	fun getFileNameWithoutExtension(f: KFile): String? {
```
Comment (alex-osm) [inline OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/util/KAlgorithms.kt:193]: Better to move these methods to KFile

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/util/KAlgorithms.kt [hunk 6]
```diff
@@ -190,4 +190,17 @@ object KAlgorithms {
...
+		return getFileNameWithoutExtension(f.name())
+	}
+
+	fun getFileNameWithoutExtension(name: String?): String? {
+		if (name != null) {
```
Comment (alex-osm) [inline OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/util/KAlgorithms.kt:198]: `return name?.substringBeforeLast(".")`

**Pull request merged**